go/arrow/csv/transformer.go

```diff
@@ -215,6 +216,15 @@ func (w *Writer) transformColToStringArr(typ arrow.DataType, col arrow.Array) []
 				res[i] = w.nullValue
 			}
 		}
+	case arrow.BinaryDataType:
+		arr := col.(*array.Binary)
```

Comment:
you can't assume this, it could also be `*array.LargeBinary` or otherwise. If you don't want to duplicate the code and create cases for both `LargeBinary` and `Binary` separately, you can create and use an interface which contains `Value(int) []byte` which would cover both `Binary` and `LargeBinary` types.

Reply:
Good catch. Should be fine now as handling only `BinaryType` in this PR (similar to how StringType only handled now)